fs/copy.py

```diff
@@ -101,13 +155,14 @@ def copy_dir(src_fs, src_path, dst_fs, dst_path, walker=None):
         in ``src_fs``. Set this if you only want to consider a sub-set
         of the resources in ``src_fs``.
     :type walker: :class:`~fs.walk.Walker`
+    :returns: a list with the copied files dst_path.
```

Comment:
I add this feature exactly to support my use case. Once you have copied some files, you may want to know which files you copied, to do something on top of them. 
In my case, i have  a data acquisition pipeline, where the copy file is just the first step. After the copy, i need to execute actions on the files, and of course know which files have been copies is super useful!

Said that, I didn’t think to the fact the list of copied files can grow enormously.
IMHO the best implementation would be to pass to copy_dir_* and to copy_fs_*  a call back function. Then, is up to the user to decide what to do inside the callback. For instance one simple implementation could be just count the copied files and sum the total bytes copied. A second option could be to collect all the files in list (as I am doing now).

The signature of the callback functions could be same as copy file:

```
Def callback(src_fs, src_path, dst_fs, dst_path):
…
```

Do you like it?